File: Green/twoSum.js
function twoSum(arr, int) {
    let ans = []
    for (let i=0; i<arr.length-1; i++) {
      for (let j=0; j<arr.length-1; j++) {
        if(i!==j && arr[i]+arr[j] === int) {
          ans.push(arr[i])
        }
      }
    }
    return ans
  }
  
  let array = [3, 5, -4, 8, 11, 1, -1, 6]
  let targetSum = 10
  
  console.log(twoSum(array, targetSum))

File: leetcode/Easy/twoSum.py
# Runtime - 32.48% and Memory - 61.66%

class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        ans = []
        for i in range(0, len(nums)):
            for j in range(i+1, len(nums)):
                if nums[i] + nums[j] == target:
                    ans.append(i)
                    ans.append(j)
                    return ans


# Solved again, on another day and in another way (more optimal likely)
# Time: O()

class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        seen = {}
        for i, num in enumerate(nums):
            diff = target - num
            if diff in seen:
                return [seen[diff], i]
            seen[num] = i